### ISSUE TITLE
`vkCmdSetScissor` behavior violates vulkan spec

### ISSUE THREAD

**Issue by Jupeyy** [KhronosGroup/MoltenVK#2304]
Disclaimer:
I do not own any apple hardware and the information provided is purely based on reports from users with apple hardware.

It seems that MoltenVK violates the vulkan spec for `vkCmdSetScissor`.

MoltenVK version:
`[mvk-info] MoltenVK version 1.2.7, supporting Vulkan version 1.2.275.`

Issue:
We create two pipelines, one with [VkPipelineDynamicStateCreateInfo](https://registry.khronos.org/VulkanSC/specs/1.0-extensions/man/html/VkPipelineDynamicStateCreateInfo.html)::pDynamicStates set to 
```cpp
std::array<VkDynamicState, 2> aDynamicStates = {
	VK_DYNAMIC_STATE_VIEWPORT,
	VK_DYNAMIC_STATE_SCISSOR,
};
```

And one without dynamic states.

The vulkan spec states:
"This command sets the scissor rectangles for subsequent drawing commands when the graphics pipeline is created with VK_DYNAMIC_STATE_SCISSOR set in VkPipelineDynamicStateCreateInfo::pDynamicStates. Otherwise, this state is specified by the VkPipelineViewportStateCreateInfo::pScissors values used to create the currently active pipeline."
(https://registry.khronos.org/VulkanSC/specs/1.0-extensions/man/html/vkCmdSetScissor.html)

In other words, a pipeline without `pDynamicStates` will use the scissor state from `pScissors` and not from the `vkCmdSetScissor` call.
And this behavior seems broken in MoltenVK.

`vkCmdSetScissor` instead also applies to the pipeline without dynamic states. We confirmed this by calling `vkCmdSetScissor` for every pipeline + saw it in renderdoc.

Note I am aware that the MoltenVK version is not the current version, but neither https://github.com/KhronosGroup/MoltenVK/blob/3b9e335fe8fc8a72cf6099acbd54128889445c0a/Docs/Whats_New.md nor any issues here on GitHub seem to be related to this issue.

I hope this helps and thanks in advance.

PS: [if you want to try it out in our app](https://github.com/ddnet/ddnet/pull/8691)
